### PR TITLE
feat: share app details on social platforms

### PR DESCRIPTION
**Pull request by tanmayrauth**
Feat #533 
It will be a button in navigation drawer that will help users to share details about the application across various social media. So that their friends can also get to know about this app and download it. 

Changes: changes were made in navigation drawer of the the app.  A new share button is added.

Screenshots for the change:
device-2019-11-16-225932
device-2019-11-16-230102

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -103,4 +104,6 @@
     <string name="clipart_saved_error">Error in saving Clipart</string>
     <string name="drawer_saved_cliparts">Saved Cliparts</string>
     <string name="no_gps_enabled">No GPS Found, Please Enable GPS from Settings</string>
+
+    <string name="share_msg">Badge Magic is an Android app to control LED name badges. This app provides features to portray names, graphics and simple animations on LED badges. \nI really loved this application, even you can download it from below link \nhttps://play.google.com/store/apps/details?id=org.fossasia.badgemagic</string>
```
Comment (yashk2000) [inline app/src/main/res/values/strings.xml]: I think the "I really love this application" part is not needed over here. Please frame a better message without it.

File: app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt [hunk 2]
```diff
@@ -124,10 +124,10 @@ class DrawerActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedLi
                         }
                         R.id.share_app_details -> {
                             val shareIntent = Intent()
-                            shareIntent.action = Intent.ACTION_SEND
                             shareIntent.type   = "text/plain"
+                            shareIntent.action = Intent.ACTION_SEND
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt]: Why did you change the line of this?

File: app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt [hunk 3]
```diff
@@ -124,10 +124,10 @@ class DrawerActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedLi
...
                             shareIntent.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_msg))
-                            startActivity(Intent.createChooser(shareIntent, "Share Using"))
+                            startActivity(Intent.createChooser(shareIntent,"Share Using"))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt]: You made it worse

File: app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt [hunk 4]
```diff
@@ -122,6 +122,13 @@ class DrawerActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedLi
                         R.id.buy -> {
                             startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://sg.pslab.io")))
                         }
+                        R.id.share_app_details -> {
+                            val shareIntent = Intent()
+                            shareIntent.type = getString(R.string.share_msg)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt]: Wrong, set proper type

File: app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt [hunk 5]
```diff
@@ -122,6 +122,13 @@ class DrawerActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedLi
...
+                            shareIntent.action = Intent.ACTION_SEND
+                            shareIntent.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_msg))
+                            startActivity(Intent.createChooser(shareIntent, getString(R.string.share_msg)))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt]: Why are you setting title to `share_msg`

File: app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt (hunk 4 above)
Comment (tanmayrauth) [reply app/src/main/java/org/fossasia/badgemagic/ui/DrawerActivity.kt]: Yeah i was testing what was wrong

**Comment by tanmayrauth** [comment]
Now the travis build is fixed @yashk2000. Can you please merge it
